api/testing/restapi_test.go

```diff
@@ -0,0 +1,1528 @@
+package testing
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/libopenstorage/openstorage/api"
+	volumeclient "github.com/libopenstorage/openstorage/api/client/volume"
+	"github.com/libopenstorage/openstorage/api/server"
+	"github.com/libopenstorage/openstorage/volume"
+	"github.com/stretchr/testify/assert"
+	"go.pedge.io/dlog"
+
+	"github.com/golang/mock/gomock"
+	"github.com/kubernetes-csi/csi-test/utils"
+
+	mockcluster "github.com/libopenstorage/openstorage/cluster/mock"
+	volumedrivers "github.com/libopenstorage/openstorage/volume/drivers"
+	mockdriver "github.com/libopenstorage/openstorage/volume/drivers/mock"
+)
+
+const (
+	host       = "http://127.0.0.1"
+	mgmtPort   = 2376
+	pluginPort = 2377
+	driver     = "mock"
+	version    = "v1"
+)
+
+// Init function to setup the http server
+
+func init() {
+	startServer()
+}
+
+// contructs the base url with host and port
+func getBaseURL() string {
+	return host + ":" + strconv.Itoa(mgmtPort)
+}
+
+func startServer() {
+
+	if err := server.StartVolumeMgmtAPI(
+		driver,
+		volume.DriverAPIBase,
+		mgmtPort,
+	); err != nil {
+		dlog.Errorf("Error starting the server")
+	}
+
+	// adding sleep to avoid race condition of connection refused.
+	time.Sleep(1 * time.Second)
+}
+
+func newMocks() *testServer {
```

Comment:
This works, but it would be better to place this in testServer.go as a `newTestServer() *testServer` function.